src/full/Agda/TypeChecking/Coverage.hs

```diff
@@ -904,11 +904,16 @@ computeNeighbourhood delta1 n delta2 d pars ixs hix tel ps cps c = do
   -- Andrea 2019-07-17 propagate the Cohesion to the equation telescope
   -- TODO: should we propagate the modality in general?
   -- See also LHS checking.
-  dtype <- do
+  (dsort, dtype) <- do
          let updCoh = composeCohesion (getCohesion info)
          TelV dtel dt <- telView dtype
-         return $ abstract (mapCohesion updCoh <$> dtel) dt
-  dsort <- addContext delta1 $ reduce (getSort dtype)
+         dsort <- fromMaybe __IMPOSSIBLE__ . isSort . unEl <$> do addContext dtel $ reduce dt
+         return $ (dsort,) $ abstract (mapCohesion updCoh <$> dtel) dt
```

Comment:
@Saizan: Do you agree to this fix, that `dsort` should be the target universe of the data type?

Reply:
Yeah, that makes sense